src/test/analyzer_test.ts

```diff
@@ -523,4 +524,63 @@ suite('Analyzer', () => {
           testSeed.events.map(e => e.name), ['fired-event', 'data-changed']);
     });
   });
+
+  suite.skip('race conditions and caching', () => {
+    const wait = () =>
+        new Promise((resolve) => setTimeout(resolve, Math.random() * 30));
```

Comment:
+1 for adding tests for those known failure states

The problem with a random test like this is that if a bug is introduced, it takes a while for the test to catch it and then takes even longer to track down where the bug was introduced since the test itself is unreliable. When it does start failing, we're much more likely to ignore it and just rerun until passing (ex: wct currently has a flakey test that's been ignored or rerun for months now).

So I'm still -1 on this test. At best it will never catch anything, and at worst we are introducing a flakey, slow-to-signal, hard-to-fix test.